src/App.jsx

```diff
@@ -189,19 +202,36 @@ export function App() {
         </div>
       </section>
 
-      <section className="section">
-        <div className="container">
-          {image ? (
-            <figure className="image">
-              <img
-                src={image}
-                alt="User uploaded file"
-                style={{ maxWidth: '500px', maxHeight: '500px', width: 'auto', height: 'auto' }}
-              />
-            </figure>
-          ) : null}
-        </div>
-      </section>
+      {permissions.data.isGranted && token ? (
+        <section className="section">
+          <div className="container">
+            <div className="box">
+              <form onSubmit={handleKuboIpfsCatFileSubmit}>
+                <div className="field">
+                  <label className="label">IPFS hash(CID)</label>
+                  <input
+                    className="input"
+                    type="text"
+                    placeholder="Example: QmRjX3..."
+                    value={hash}
+                    onChange={({ target }) => setHash(target.value)}
+                  />
+                  {kuboIpfsCatFile.isError ? (
+                    <p className="help is-danger">{kuboIpfsCatFile.error.message}</p>
+                  ) : null}
+                </div>
+                <button
```

Comment:
Because `cat` has no auth we can have a simple `a href=".../cat?arg=Qm" target="_blank"` and remove almost everything